cmd/agent/app/flare.go

```diff
@@ -117,6 +135,13 @@ func readProfileData(pdata *flare.ProfileData) error {
 	return flare.CreatePerformanceProfile("trace", traceDebugURL, cpusec, pdata)
 }
 
+func readProcessAgentProfileData(pdata *flare.ProfileData) error {
```

Comment:
So we could possibly test the fact that individual failures don't fail flare collection, this can be done by mocking the `CreatePerformanceProfile` func. Beyond this basic part it's a bit challenging since you need the actual agents fully bootstrapped.

Reply:
That's fair. I will leave it up to you whether or not you want to create a test for this

Reply:
Added a couple of basic tests 🙇